cms/src/migrations/9005-new-views-for-access.ts

```diff
@@ -0,0 +1,255 @@
+                                                                  WHERE se.show_id = sh.id))))),
+                ARRAY []::character varying[]) AS roles,
+       COALESCE((SELECT array_agg(DISTINCT eu.usergroups_code) AS code
+                 FROM episodes_usergroups_download eu
+                 WHERE (eu.episodes_id IN (SELECT e.id
+                                           FROM episodes e
+                                           WHERE (e.season_id IN (SELECT se.id
+                                                                  FROM seasons se
+                                                                  WHERE se.show_id = sh.id))))),
+                ARRAY []::character varying[]) AS roles_download,
+       COALESCE((SELECT array_agg(DISTINCT eu.usergroups_code) AS code
+                 FROM episodes_usergroups_earlyaccess eu
+                 WHERE (eu.episodes_id IN (SELECT e.id
+                                           FROM episodes e
+                                           WHERE (e.season_id IN (SELECT se.id
+                                                                  FROM seasons se
+                                                                  WHERE se.show_id = sh.id))))),
+                ARRAY []::character varying[]) AS roles_earlyaccess
+FROM shows sh;
+
+create or replace view shows_availability(id, published, available_from, available_to) as
+SELECT sh.id,
+       sh.status::text = 'published'::text                                             AS published,
+       COALESCE(sh.available_from, '1800-01-01 00:00:00'::timestamp without time zone) AS available_from,
+       COALESCE(sh.available_to, '3000-01-01 00:00:00'::timestamp without time zone)   AS available_to
+FROM shows sh;
+
+create or replace view shows_access_view
+            (id, published, available_from, available_to, usergroups, usergroups_downloads, usergroups_earlyaccess) as
+SELECT sh.id,
+       a.published,
+       a.available_from,
+       a.available_to,
+       r.roles             AS usergroups,
+       r.roles_download    AS usergroups_downloads,
+       r.roles_earlyaccess AS usergroups_earlyaccess
+FROM shows sh
+         LEFT JOIN shows_availability a ON a.id = sh.id
+         LEFT JOIN shows_roles r ON r.id = sh.id;
+`;
+
+const shows_materialized_view_sql = `
+create materialized view if not exists shows_access as
+SELECT shows_access_view.id,
+       shows_access_view.published,
+       shows_access_view.available_from,
+       shows_access_view.available_to,
+       shows_access_view.usergroups,
+       shows_access_view.usergroups_downloads,
+       shows_access_view.usergroups_earlyaccess
+FROM shows_access_view;
+
+create unique index if not exists shows_access_idx
+    on shows_access (id);
+`
+
+const update_access_function = `
+create or replace function update_access(view character varying) returns boolean
+    language plpgsql
+as
+$$
+DECLARE
+    lr timestamptz;
+BEGIN
```

Comment:
I think you are missing `SECURITY DEFINER` or something like that here. The reason is that the view can only be refreshed by the user who created it and by having that on the function it is executed as the user who defined it.